sdk/src/math/protectedMakerParams.ts

```diff
@@ -0,0 +1,34 @@
+import { MarketTypeStr, BN, ProtectMakerParamsMap } from "..";
+import { PerpMarketAccount, ProtectedMakerParams } from "../types";
+
+export function getProtectedMakerParams(
+    perpMarket: PerpMarketAccount,
+): ProtectedMakerParams {
+    let dynamicOffset;
+    if (perpMarket.protectedMakerDynamicDivisor > 0) {
+        dynamicOffset = BN.max(perpMarket.amm.oracleStd, perpMarket.amm.markStd).divn(perpMarket.protectedMakerDynamicDivisor);
+    } else {
+        dynamicOffset =  0;
+    }
+    
+    return {
+        tickSize: perpMarket.amm.orderTickSize,
+        limitPriceDivisor: perpMarket.protectedMakerLimitPriceDivisor,
+        dynamicOffset: dynamicOffset,
+    }
+}
+
+export function getProtectedMakerParamsMap(
+    perpMarkets: PerpMarketAccount[],
+): ProtectMakerParamsMap {
+    const map = {
+        perp: new Map<number, ProtectedMakerParams>(),
+        spot: new Map<number, ProtectedMakerParams>(),
+    }
+    for (const perpMarket of perpMarkets) {
+        const marketIndex = perpMarket.marketIndex;
+        const protectedMakerParams = getProtectedMakerParams(perpMarket);
+        map.perp.set(marketIndex, protectedMakerParams);
+    }
```

Comment:
dont know if you should also populate this with the spot market indexes, and just set it to undefined since the map is read on insert and 'spot' will be undefined https://github.com/drift-labs/protocol-v2/blob/eed2244669da421df6b58387af776ca4b67db20f/sdk/src/dlob/DLOB.ts#L403

Reply:
i was thinking we'd return undefined intentionally to signal to not apply buffer